bin/outrigger-client

```diff
@@ -101,34 +104,47 @@ function read_inputs(input_args) {
-    // Open a websocket connection over the rendezvous channel
-    var ws = new WebSocket(juttled_noapi_ws_url + '/rendezvous/' + topic);
-    ws.on('open', function() {
-        // Bundle the program locally and push the bundle over the websocket.
-        var bundler = new JuttleBundler();
-        bundler.bundle(filename)
-            .then(function(bundle) {
-                ws.send(JSON.stringify(bundle));
-            })
-            .catch(function(err) {
-                // If it's a juttle error containing a program
-                // location, print the error in context.
-                if (err.code === 'JS-JUTTLE-ERROR') {
-                    console.error(cli_errors.show_in_context({err: err.info.err,
-                                                              program: err.info.bundle.program,
-                                                              modules: err.info.bundle.modules,
-                                                              filename: filename}));
-                } else {
-                    console.error("ERROR", err.message);
-                }
-            })
-            .finally(function() {
-                ws.close();
-            });
+    return new Promise(function (resolve, reject) {
+
+        console.log("Bundling program " + filename + " and pushing to rendezvous topic " + topic + "...");
+        // Open a websocket connection over the rendezvous channel
+        var ws = new WebSocket(juttled_noapi_ws_url + '/rendezvous/' + topic);
+        ws.on('open', function() {
+            // Bundle the program locally and push the bundle over the websocket.
+            var bundler = new JuttleBundler();
+            bundler.bundle(filename)
+                .then(function(bundle) {
+                    return new Promise(function (resolve, reject) {
+                        ws.send(JSON.stringify(bundle), resolve);
+                    });
+                })
+                .catch(function(err) {
+                    // If it's a juttle error containing a program
+                    // location, print the error in context.
+                    if (err.code === 'JS-JUTTLE-ERROR') {
+                        console.error(cli_errors.show_in_context({err: err.info.err,
+                                                                  program: err.info.bundle.program,
+                                                                  modules: err.info.bundle.modules,
+                                                                  filename: filename}));
+                    } else {
+                        console.error("ERROR", err.message);
+                    }
+                })
+                .finally(function() {
+                    ws.close();
+                    resolve();
+                });
+        });
     });
 }
 
+function get_file_hash(path) {
+    var data = fs.readFileSync(path);
```

Comment:
I think for a cli using sync is fine, but this is just one mans opinion. I could easily promisify it.